modules/assim.sequential/R/sda.enkf.R

```diff
@@ -482,83 +481,73 @@ sda.enkf <- function(settings, obs.mean, obs.cov, IC = NULL, Q = NULL) {
         }
         
         #### These vectors are used to categorize data based on censoring from the interval matrix
-        x.ind <- matrix(NA, nens, ncol(X)) ; x.censored <- x.ind
-        for(n in 1:nens){
-          x.ind[n,] <- as.numeric(X[n,] > intervalX[,1])
-          x.censored[n,] <- as.numeric(ifelse(X[n,] > intervalX[,1], X[n,], 0))
-        }
+        x.ind <- as.numeric(mu.f > intervalX[,1])
+        x.censored <- as.numeric(ifelse(mu.f > intervalX[,1], mu.f, 0)) #probably not needed until different data
         
-        if(t == 1){ #| length(x.ind[1,]) > mu.f
-          #y.obs = Y.dat[1,]
-          constants.tobit2space = list(N = ncol(X), nens = nens)
-          data.tobit2space = list(interval = intervalX,
-                                  y.ind = x.ind,
-                                  y.censored = x.censored,
-                                  aq = diag(ncol(X))*ncol(X), 
-                                  bq = ncol(X),
-                                  mu.prior = colMeans(X), #cheating? basically gives us back means
-                                  cov.prior = diag(ncol(X)))
-          
-          inits.tobit2space = list(pf = diag(ncol(X)), muf = rep(0,ncol(X))) #
+        if(t == 1){
+          #The purpose of this step is to impute data for mu.f 
+          #where there are zero values so that 
+          #mu.f is in 'tobit space' in the full model
+          constants.tobit2space = list(N = length(mu.f),
+                                       muf = mu.f,
+                                       pf = solve(Pf))
+          data.tobit2space = list(y.ind = x.ind,
+                                  y.censored = x.censored)
           #set.seed(0)
           #ptm <- proc.time()
           tobit2space_pred <- nimbleModel(tobit2space.model, data = data.tobit2space,
-                                          constants = constants.tobit2space, inits = inits.tobit2space)
+                                          constants = constants.tobit2space)
```

Comment:
would nimble convergence be quicker if we retained the specification of inits? I'd guess that sample mean and cov would be a decent starting point

Reply:
according to the nimble manual 'Nodes following a dinterval (or dconstraint) distribution should normally be set as data with known values. Otherwise, the node may be simulated during initialization in some algorithms (e.g.,MCMC) and thereby establish a permanent, perhaps unintended, constraint.' So, I took out the inits to avoid that problem.